CJS 4.8.0 - can't compile with Clang (similar issue was fixed in GJS)

I cannot compile the latest version of CJS 4.8.0 using the Clang 11 compiler on OpenMandriva Cooker.

At build time I see this error: 

```
../gi/gtype.cpp:58:1: error: an attribute list cannot appear here
DEBUG util.py:623:  GJS_DEFINE_PRIV_FROM_JS(void, gjs_gtype_class);
DEBUG util.py:623:  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
DEBUG util.py:623:  ../cjs/jsapi-class.h:80:23: note: expanded from macro 'GJS_DEFINE_PRIV_FROM_JS'
DEBUG util.py:623:      GJS_ALWAYS_INLINE [[nodiscard]] [[maybe_unused]] static inline bool        \
DEBUG util.py:623:                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
DEBUG util.py:623:  ../gi/gtype.cpp:58:1: error: an attribute list cannot appear here
DEBUG util.py:623:  GJS_DEFINE_PRIV_FROM_JS(void, gjs_gtype_class);
DEBUG util.py:623:  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
DEBUG util.py:623:  ../cjs/jsapi-class.h:84:23: note: expanded from macro 'GJS_DEFINE_PRIV_FROM_JS'
DEBUG util.py:623:      GJS_ALWAYS_INLINE [[nodiscard]] static inline type* priv_from_js(          \
DEBUG util.py:623:                        ^~~~~~~~~~~~~
DEBUG util.py:623:  2 errors generated.
```
Full build log: https://file-store.openmandriva.org/api/v1/file_stores/6c2f1adb715237cc558714381fa856a5d913c2f2.log?show=true

Worth to add, everything compiles correctly when I use the GCC compiler. But the thing is, we try our best to compile everything that we can with Clang.

I would like to add that the same error (or a similar one) also appeared in the case of GJS 1.66 compiled with Clang. Here is a link to upstream issues and a link below to the fix for this problem.

GJS 1.66 upstream issue: https://gitlab.gnome.org/GNOME/gjs/-/issues/347
GJS 1.66 fix issue: https://gitlab.gnome.org/GNOME/gjs/-/merge_requests/499

And it looks like this problem has already been fixed in GJS 1.66.1.

--- a/cjs/jsapi-class.h
+++ b/cjs/jsapi-class.h
@@ -77,11 +77,11 @@ bool gjs_define_property_dynamic(JSContext       *cx,
  *                              do_base_typecheck and priv_from_js
  */
 #define GJS_DEFINE_PRIV_FROM_JS(type, klass)                                   \
-    GJS_ALWAYS_INLINE [[nodiscard]] [[maybe_unused]] static inline bool        \
+    [[nodiscard]] [[maybe_unused]] GJS_ALWAYS_INLINE static inline bool        \
     do_base_typecheck(JSContext* cx, JS::HandleObject obj, bool throw_error) { \
         return gjs_typecheck_instance(cx, obj, &klass, throw_error);           \
     }                                                                          \
-    GJS_ALWAYS_INLINE [[nodiscard]] static inline type* priv_from_js(          \
+    [[nodiscard]] GJS_ALWAYS_INLINE static inline type* priv_from_js(          \
         JSContext* cx, JS::HandleObject obj) {                                 \
         return static_cast<type*>(                                             \
             JS_GetInstancePrivate(cx, obj, &klass, nullptr));                  \

--- a/cjs/jsapi-util-args.h
+++ b/cjs/jsapi-util-args.h
@@ -43,7 +43,7 @@
 #include "cjs/jsapi-util.h"
 #include "cjs/macros.h"
 
-GJS_ALWAYS_INLINE [[nodiscard]] static inline bool check_nullable(
+[[nodiscard]] GJS_ALWAYS_INLINE static inline bool check_nullable(
     const char*& fchar, const char*& fmt_string) {
     if (*fchar != '?')
         return false;

